test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,167 @@
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Unknown"}'
+        values: '1x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-4", phase="Succeeded"}'
+        values: '0x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-4", pod="pod-1"}'
+        values: '1x0 0x13'
+
+      # Pod is in Succeeded state and scheduled, so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Unknown"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-5", phase="Succeeded"}'
+        values: '1x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-5", pod="pod-1"}'
+        values: '0x14'
+
+      # Pod is in the Running state for the first 14 min, then it's in the Failed state
+      # for the last 1 min and unscheduled, so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-7", phase="Running"}'
+        values: '1x13 0'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-7", phase="Failed"}'
+        values: '0x13 1'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-7", pod="pod-1"}'
+        values: '0x14'
+
+      # Pod is in the Failed state and scheduled, but it has a namespace that ends
+      # with 'tenant' so it's ignored.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="prod-tenant", phase="Failed"}'
+        values: '1x14'
+      - series: 'kube_pod_status_unschedulable{namespace="prod-tenant", pod="pod-1"}'
+        values: '0x14'
+
+      # Pod is in the Pending state and scheduled, but it has a namespace that starts
+      # with 'openshift' so it's ignored.
+      - series: 'kube_pod_status_phase{pod="pod-2", namespace="openshift-prod-test", phase="Pending"}'
+        values: '1x14'
+      - series: 'kube_pod_status_unschedulable{namespace="openshift-prod-test", pod="pod-2"}'
+        values: '0x14'
+
+      # Pod is in the Pending state and scheduled, but it has a namespace that starts
+      # with 'kube' so it's ignored.
+      - series: 'kube_pod_status_phase{pod="pod-3", namespace="kube-test", phase="Unknown"}'
+        values: '1x14'
+      - series: 'kube_pod_status_unschedulable{namespace="kube-test", pod="pod-3"}'
+        values: '0x14'
+
+      # Pod is in the Pending state and scheduled, but it has a namespace 'default
```

Comment:
```suggestion
      # Pod is in the Pending state and scheduled, but it has a namespace 'default'
```

Reply:
Thank you for correcting all my tiny errors @yftacherzog